components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/ext/APIManagerExtensionHandler.java

```diff
@@ -71,9 +75,21 @@ public boolean mediate(MessageContext messageContext, String direction) {
     public boolean handleRequest(MessageContext messageContext) {
         Timer.Context context = startMetricTimer(DIRECTION_IN);
         long executionStartTime = System.nanoTime();
+        TracingSpan requestMediationSpan = null;
+        Boolean tracingEnabled = Boolean.valueOf(ServiceReferenceHolder.getInstance().getAPIManagerConfiguration()
+                .getFirstProperty(APIMgtGatewayConstants.TRACING_ENABLED));
+        if (tracingEnabled) {
+            TracingSpan responseLatencySpan =
+                    (TracingSpan) messageContext.getProperty(APIMgtGatewayConstants.RESPONSE_LATENCY);
+            TracingTracer tracer = Util.getGlobalTracer();
+            requestMediationSpan = Util.startSpan(APIMgtGatewayConstants.REQUEST_MEDIATION, responseLatencySpan, tracer);
+        }
         try {
             return mediate(messageContext, DIRECTION_IN);
         } finally {
+            if (tracingEnabled) {
```

Comment:
You can also add the errors related information on the spans with error tag. https://github.com/opentracing/specification/blob/master/semantic_conventions.md. With those you can also provide the error related information by using the proper tags. Therefore, use the catch clause and set the error tags correctly.

Reply:
fixed